src/components/slider/draw.js

```diff
@@ -0,0 +1,507 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var d3 = require('d3');
+
+var Plotly = require('../../plotly');
+var Plots = require('../../plots/plots');
+var Lib = require('../../lib');
+var Color = require('../color');
+var Drawing = require('../drawing');
+var svgTextUtils = require('../../lib/svg_text_utils');
+var anchorUtils = require('../legend/anchor_utils');
+
+var constants = require('./constants');
+
+
+module.exports = function draw(gd) {
+    var fullLayout = gd._fullLayout,
+        sliderData = makeSliderData(fullLayout);
+
+    // draw a container for *all* sliders:
+    var sliders = fullLayout._infolayer
+        .selectAll('g.' + constants.containerClassName)
+        .data(sliderData.length > 0 ? [0] : []);
+
+    sliders.enter().append('g')
+        .classed(constants.containerClassName, true)
+        .style('cursor', 'ew-resize');
+
+    sliders.exit().remove();
+
+    // If no more sliders, clear the margisn:
+    if(sliders.exit().size()) clearPushMargins(gd);
```

Comment:
When sliders disappear, it also needs to detach event subscriptions. That's very important! We should think about the proper way for components to do that in general.